lib/asls/server.ex

```diff
@@ -73,55 +72,29 @@ defmodule AssemblyScriptLS.Server do
         }
     end
 
-    {:reply, payload, state}
+    # FIXME: Ensure that runtime dependencies are met
+    # before blindly initializing the server
+    {:ok, rt} = Runtime.ensure(params[:rootUri])
+    {:reply, payload, %{state | runtime: rt}}
   end
 
   @impl true
   def handle_call({:notification, %Notification{method: "initialized"}}, _from, state) do
     {:reply, :ok, %{state | initialized: true}}
   end
 
-  @impl true
-  def handle_call({:notification, %Notification{method: "workspace/didChangeConfiguration"}}, _from, state) do
-    cond do
-      state.building? ->
-        {:reply, :ok, %{state | rebuild?: true}}
-      true ->
-        task = @builder.perform(%{root_uri: state.root_uri})
-        {:reply, :ok, %{state | build_ref: task.ref, building?: true, rebuild?: false}}
-    end
-  end
-
   @impl true
   def handle_call({:notification, %Notification{method: "textDocument/didOpen"} = req}, from, state) do
     GenServer.reply(from, :ok)
-    uri = req.params[:textDocument].uri
-
-    @rpc.notify("textDocument/publishDiagnostics", %{
-      uri: uri,
-      diagnostics: state.diagnostics[uri] || []
-    })
-
-    state =
-      cond do
-        Enum.find(state.documents, &(&1 == uri)) ->
-          state
-        true ->
-          %{state | documents: [uri | state.documents]}
-      end
-
+    state = enqueue_analysis(state, req.params[:textDocument].uri)
```

Comment:
Report diagnostics here in case the document was opened, closed, and opened again.